### ISSUE TITLE
Plugin call order and extension to reader preferences are broken

### ISSUE THREAD

**Issue by goanpeca** [napari/napari#4175]
## 🐛 Bug

Opening the preferences and changing anything in the plugin page will result in two issues:

-  The call order is storing duplicates for plugins that provide several hooks because the plugin name is stored and not the name of the reader
- The extension to reader widget did not provide a `value()` method which will result in a ` AttributeError: 'ObjectSchemaWidget' object has no attribute 'state'`

## To Reproduce

0. Install `napari-brainglobe-io` which provides several readers
1. Open the preferences and change anything in the plugin page (call order for readers for example)
2. See an error `AttributeError: 'ObjectSchemaWidget' object has no attribute 'state'`
3. Close napari, open napari, it will fail to start with a `ValueError: repeated item in order`

## Environment

 - Please copy and paste the information at napari info option in help menubar here:

 - Any other relevant information:

## Additional context

This bug was found while working on https://github.com/napari/napari/pull/4074

**Comment by DragaDoncila** [comment]
I think I know what's going on here and how to fix, will take a look and should have a PR up today

**Comment by DragaDoncila** [comment]
Ok so I've reproduced and got to the bottom of the issues here:

> The extension to reader widget did not provide a value() method which will result in a AttributeError: 'ObjectSchemaWidget' object has no attribute 'state'

I've opened #4179 to address this. This issue is separate from:

> The call order is storing duplicates for plugins that provide several hooks because the plugin name is stored and not the name of the reader

This has been broken since at least 0.4.10 (earliest I checked, though I'm pretty sure it never worked) for plugins which declare multiple readers. To clarify, sorting is fine, but closing and reopening napari leads to `ValueError: repeated item in order` and napari doesn't open. 

As @goanpeca has noted, *one of* this issue OR #4177 is critical to fix for the bundled app, because bundled app users don't have a way of resetting preferences.

I've drafted up #4180 to address the repeated item issue. This fix would be invisible to the user but as mentioned in the PR, would require a settings reset since we're changing the persisted value. The entire plugin call order sorter will be removed once we switch over to `npe2` following #4015 - we should be ready to do this once #4102 (or some version of it) and #4146 are addressed, as this will give users explicit control over reader choice in all cases. @nclack and @tlambert03 can weigh in more here. Also pinging @jni as this is relevant to release.

Given this behaviour was always broken and we're planning on removing it anyway we may just want to address #4177? I'm also not the most familiar with the original plugin engine so maybe what I've done in #4180 is a total hack 🤷‍♀️ . 

There is a *third* issue that I've realized is a problem, which is that we'll run into the same issue assigning preferences for npe2 plugins with multiple readers - I will open a separate issue to track this though.
